central/graphql/resolvers/loaders/nodes.go

```diff
@@ -58,16 +60,38 @@ type nodeLoaderImpl struct {
 
 // FromIDs loads a set of nodes from a set of ids.
 func (ndl *nodeLoaderImpl) FromIDs(ctx context.Context, ids []string) ([]*storage.Node, error) {
-	nodes, err := ndl.load(ctx, ids)
+	nodes, err := ndl.load(ctx, ids, false)
 	if err != nil {
 		return nil, err
 	}
 	return nodes, nil
 }
 
-// FromID loads a node from an ID.
+// FromID loads a node from an ID. Does not load scan components and vulnerabilities when Postgres is enabled.
 func (ndl *nodeLoaderImpl) FromID(ctx context.Context, id string) (*storage.Node, error) {
-	nodes, err := ndl.load(ctx, []string{id})
+	nodes, err := ndl.load(ctx, []string{id}, false)
+	if err != nil {
+		return nil, err
+	}
+	return nodes[0], nil
+}
+
+// FullNodeWithID loads full node from an ID.
+func (ndl *nodeLoaderImpl) FullNodeWithID(ctx context.Context, id string) (*storage.Node, error) {
+	node, err := ndl.FromID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	// Load the full node if full scan is not available.
+	if node.GetComponents() == 0 || len(node.GetScan().GetComponents()) > 0 {
+		return node, nil
+	}
+
+	ndl.lock.Lock()
+	delete(ndl.loaded, id)
```

Comment:
I did not see codes to expire entries in loaded, will it become full node eventually? Of course, that is not introduced in this PR.

Reply:
Yes it will become full node eventually, and the next time same node it queried, it will not fetch full node again.